from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(1931, 1087)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.widget = QtWidgets.QWidget(self.centralwidget)
        self.widget.setGeometry(QtCore.QRect(330, 0, 1581, 971))
        self.widget.setStyleSheet("background-color:white")
        self.widget.setObjectName("widget")
        self.widget_2 = QtWidgets.QWidget(self.centralwidget)
        self.widget_2.setGeometry(QtCore.QRect(0, 0, 321, 981))
        self.widget_2.setStyleSheet("background-color:white")
        self.widget_2.setObjectName("widget_2")
        self.widget_3 = QtWidgets.QWidget(self.widget_2)
        self.widget_3.setGeometry(QtCore.QRect(10, 10, 301, 61))
        self.widget_3.setStyleSheet("background-color:lightblue")
        self.widget_3.setObjectName("widget_3")
        self.label = QtWidgets.QLabel(self.widget_3)
        self.label.setGeometry(QtCore.QRect(90, 10, 111, 31))
        font = QtGui.QFont()
        font.setFamily("Times New Roman")
        font.setPointSize(14)
        font.setBold(False)
        font.setWeight(50)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.scrollArea = QtWidgets.QScrollArea(self.widget_2)
        self.scrollArea.setGeometry(QtCore.QRect(10, 80, 301, 811))
        self.scrollArea.setWidgetResizable(True)
        self.scrollArea.setObjectName("scrollArea")
        self.scrollAreaWidgetContents = QtWidgets.QWidget()
        self.scrollAreaWidgetContents.setGeometry(QtCore.QRect(0, 0, 299, 809))
        self.scrollAreaWidgetContents.setObjectName("scrollAreaWidgetContents")
        self.scrollArea.setWidget(self.scrollAreaWidgetContents)
        self.widget_4 = QtWidgets.QWidget(self.widget_2)
        self.widget_4.setGeometry(QtCore.QRect(10, 900, 291, 71))
        self.widget_4.setStyleSheet("background-color:lightblue")
        self.widget_4.setObjectName("widget_4")
        self.pushButton = QtWidgets.QPushButton(self.widget_4)
        self.pushButton.setGeometry(QtCore.QRect(60, 10, 161, 51))
        font = QtGui.QFont()
        font.setFamily("Times New Roman")
        font.setPointSize(14)
        font.setBold(True)
        font.setWeight(75)
        self.pushButton.setFont(font)
        self.pushButton.setStyleSheet("background-color: lightgreen")
        self.pushButton.setObjectName("pushButton")
        MainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 1931, 26))
        self.menubar.setObjectName("menubar")
        self.menuFile_F = QtWidgets.QMenu(self.menubar)
        self.menuFile_F.setObjectName("menuFile_F")
        self.menuEdit_E = QtWidgets.QMenu(self.menubar)
        self.menuEdit_E.setObjectName("menuEdit_E")
        self.menuView_V = QtWidgets.QMenu(self.menubar)
        self.menuView_V.setObjectName("menuView_V")
        self.menuHelp_H = QtWidgets.QMenu(self.menubar)
        self.menuHelp_H.setObjectName("menuHelp_H")
        MainWindow.setMenuBar(self.menubar)
        self.toolBar = QtWidgets.QToolBar(MainWindow)
        self.toolBar.setEnabled(True)
        self.toolBar.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.toolBar.setIconSize(QtCore.QSize(40, 40))
        self.toolBar.setObjectName("toolBar")
        MainWindow.addToolBar(QtCore.Qt.TopToolBarArea, self.toolBar)
        self.toolBar_2 = QtWidgets.QToolBar(MainWindow)
        self.toolBar_2.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.toolBar_2.setIconSize(QtCore.QSize(40, 40))
        self.toolBar_2.setObjectName("toolBar_2")
        MainWindow.addToolBar(QtCore.Qt.TopToolBarArea, self.toolBar_2)
        self.actionNew_Project_Ctrl_N = QtWidgets.QAction(MainWindow)
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap("Icons/New.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionNew_Project_Ctrl_N.setIcon(icon)
        self.actionNew_Project_Ctrl_N.setObjectName("actionNew_Project_Ctrl_N")
        self.actionOpen_Project_Ctrl_O = QtWidgets.QAction(MainWindow)
        icon1 = QtGui.QIcon()
        icon1.addPixmap(QtGui.QPixmap("Icons/Open.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionOpen_Project_Ctrl_O.setIcon(icon1)
        self.actionOpen_Project_Ctrl_O.setObjectName("actionOpen_Project_Ctrl_O")
        self.actionSave_Project_Ctrl_S = QtWidgets.QAction(MainWindow)
        icon2 = QtGui.QIcon()
        icon2.addPixmap(QtGui.QPixmap("Icons/Save.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionSave_Project_Ctrl_S.setIcon(icon2)
        self.actionSave_Project_Ctrl_S.setObjectName("actionSave_Project_Ctrl_S")
        self.actionSettings = QtWidgets.QAction(MainWindow)
        icon3 = QtGui.QIcon()
        icon3.addPixmap(QtGui.QPixmap("Icons/Settings.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionSettings.setIcon(icon3)
        self.actionSettings.setObjectName("actionSettings")
        self.actionExit_X = QtWidgets.QAction(MainWindow)
        icon4 = QtGui.QIcon()
        icon4.addPixmap(QtGui.QPixmap("Icons/Close.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionExit_X.setIcon(icon4)
        self.actionExit_X.setObjectName("actionExit_X")
        self.actionUndo_Ctrl_Z = QtWidgets.QAction(MainWindow)
        icon5 = QtGui.QIcon()
        icon5.addPixmap(QtGui.QPixmap("Icons/Undo.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionUndo_Ctrl_Z.setIcon(icon5)
        self.actionUndo_Ctrl_Z.setObjectName("actionUndo_Ctrl_Z")
        self.actionRedo_Ctrl_Y = QtWidgets.QAction(MainWindow)
        icon6 = QtGui.QIcon()
        icon6.addPixmap(QtGui.QPixmap("Icons/Redo.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionRedo_Ctrl_Y.setIcon(icon6)
        self.actionRedo_Ctrl_Y.setObjectName("actionRedo_Ctrl_Y")
        self.actionCut_Ctrl_X = QtWidgets.QAction(MainWindow)
        icon7 = QtGui.QIcon()
        icon7.addPixmap(QtGui.QPixmap("Icons/Cut.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionCut_Ctrl_X.setIcon(icon7)
        self.actionCut_Ctrl_X.setObjectName("actionCut_Ctrl_X")
        self.actionCopy_Ctrl_C = QtWidgets.QAction(MainWindow)
        icon8 = QtGui.QIcon()
        icon8.addPixmap(QtGui.QPixmap("Icons/Copy.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionCopy_Ctrl_C.setIcon(icon8)
        self.actionCopy_Ctrl_C.setObjectName("actionCopy_Ctrl_C")
        self.actionPaste_Ctrl_V = QtWidgets.QAction(MainWindow)
        self.actionPaste_Ctrl_V.setCheckable(True)
        self.actionPaste_Ctrl_V.setChecked(True)
        icon9 = QtGui.QIcon()
        icon9.addPixmap(QtGui.QPixmap("Icons/Paste.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionPaste_Ctrl_V.setIcon(icon9)
        self.actionPaste_Ctrl_V.setObjectName("actionPaste_Ctrl_V")
        self.actionDelete = QtWidgets.QAction(MainWindow)
        self.actionDelete.setCheckable(True)
        self.actionDelete.setChecked(True)
        icon10 = QtGui.QIcon()
        icon10.addPixmap(QtGui.QPixmap("Icons/Delete.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionDelete.setIcon(icon10)
        self.actionDelete.setObjectName("actionDelete")
        self.actionSelect_All_Ctrl_A = QtWidgets.QAction(MainWindow)
        self.actionSelect_All_Ctrl_A.setCheckable(True)
        self.actionSelect_All_Ctrl_A.setChecked(True)
        icon11 = QtGui.QIcon()
        icon11.addPixmap(QtGui.QPixmap("Icons/Select All.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionSelect_All_Ctrl_A.setIcon(icon11)
        self.actionSelect_All_Ctrl_A.setObjectName("actionSelect_All_Ctrl_A")
        self.actionDevices = QtWidgets.QAction(MainWindow)
        icon12 = QtGui.QIcon()
        icon12.addPixmap(QtGui.QPixmap("Icons/Electronic Devices.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionDevices.setIcon(icon12)
        self.actionDevices.setObjectName("actionDevices")
        self.actionHow_To_Use = QtWidgets.QAction(MainWindow)
        icon13 = QtGui.QIcon()
        icon13.addPixmap(QtGui.QPixmap("Icons/How to Use.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionHow_To_Use.setIcon(icon13)
        self.actionHow_To_Use.setObjectName("actionHow_To_Use")
        self.actionReport_Issue = QtWidgets.QAction(MainWindow)
        icon14 = QtGui.QIcon()
        icon14.addPixmap(QtGui.QPixmap("Icons/Report.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionReport_Issue.setIcon(icon14)
        self.actionReport_Issue.setObjectName("actionReport_Issue")
        self.actionCheck_For_Updaes = QtWidgets.QAction(MainWindow)
        icon15 = QtGui.QIcon()
        icon15.addPixmap(QtGui.QPixmap("Icons/Update.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionCheck_For_Updaes.setIcon(icon15)
        self.actionCheck_For_Updaes.setObjectName("actionCheck_For_Updaes")
        self.actionAbout = QtWidgets.QAction(MainWindow)
        icon16 = QtGui.QIcon()
        icon16.addPixmap(QtGui.QPixmap("Icons/About.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionAbout.setIcon(icon16)
        self.actionAbout.setObjectName("actionAbout")
        self.actionReset_Crtl_R = QtWidgets.QAction(MainWindow)
        icon17 = QtGui.QIcon()
        icon17.addPixmap(QtGui.QPixmap("Icons/Reset.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionReset_Crtl_R.setIcon(icon17)
        self.actionReset_Crtl_R.setObjectName("actionReset_Crtl_R")
        self.actionHome_Ctrl_H = QtWidgets.QAction(MainWindow)
        icon18 = QtGui.QIcon()
        icon18.addPixmap(QtGui.QPixmap("Icons/Home.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionHome_Ctrl_H.setIcon(icon18)
        self.actionHome_Ctrl_H.setObjectName("actionHome_Ctrl_H")
        self.actionShortcut_File_Menu = QtWidgets.QAction(MainWindow)
        icon19 = QtGui.QIcon()
        icon19.addPixmap(QtGui.QPixmap("Icons/Shortcut File Menu.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionShortcut_File_Menu.setIcon(icon19)
        self.actionShortcut_File_Menu.setObjectName("actionShortcut_File_Menu")
        self.actionShortcut_Edit_Menu = QtWidgets.QAction(MainWindow)
        icon20 = QtGui.QIcon()
        icon20.addPixmap(QtGui.QPixmap("Icons/Shortcut Edit Menu.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionShortcut_Edit_Menu.setIcon(icon20)
        self.actionShortcut_Edit_Menu.setObjectName("actionShortcut_Edit_Menu")
        self.menuFile_F.addAction(self.actionHome_Ctrl_H)
        self.menuFile_F.addSeparator()
        self.menuFile_F.addAction(self.actionNew_Project_Ctrl_N)
        self.menuFile_F.addSeparator()
        self.menuFile_F.addAction(self.actionOpen_Project_Ctrl_O)
        self.menuFile_F.addSeparator()
        self.menuFile_F.addAction(self.actionSave_Project_Ctrl_S)
        self.menuFile_F.addSeparator()
        self.menuFile_F.addAction(self.actionSettings)
        self.menuFile_F.addSeparator()
        self.menuFile_F.addAction(self.actionExit_X)
        self.menuEdit_E.addAction(self.actionUndo_Ctrl_Z)
        self.menuEdit_E.addAction(self.actionRedo_Ctrl_Y)
        self.menuEdit_E.addAction(self.actionReset_Crtl_R)
        self.menuEdit_E.addSeparator()
        self.menuEdit_E.addAction(self.actionCut_Ctrl_X)
        self.menuEdit_E.addAction(self.actionCopy_Ctrl_C)
        self.menuEdit_E.addAction(self.actionPaste_Ctrl_V)
        self.menuEdit_E.addSeparator()
        self.menuEdit_E.addAction(self.actionDelete)
        self.menuEdit_E.addAction(self.actionSelect_All_Ctrl_A)
        self.menuView_V.addAction(self.actionShortcut_File_Menu)
        self.menuView_V.addAction(self.actionShortcut_Edit_Menu)
        self.menuHelp_H.addAction(self.actionDevices)
        self.menuHelp_H.addAction(self.actionHow_To_Use)
        self.menuHelp_H.addSeparator()
        self.menuHelp_H.addAction(self.actionReport_Issue)
        self.menuHelp_H.addAction(self.actionCheck_For_Updaes)
        self.menuHelp_H.addSeparator()
        self.menuHelp_H.addAction(self.actionAbout)
        self.menubar.addAction(self.menuFile_F.menuAction())
        self.menubar.addAction(self.menuEdit_E.menuAction())
        self.menubar.addAction(self.menuView_V.menuAction())
        self.menubar.addAction(self.menuHelp_H.menuAction())
        self.toolBar.addAction(self.actionHome_Ctrl_H)
        self.toolBar.addAction(self.actionNew_Project_Ctrl_N)
        self.toolBar.addAction(self.actionOpen_Project_Ctrl_O)
        self.toolBar.addAction(self.actionSave_Project_Ctrl_S)
        self.toolBar_2.addAction(self.actionReset_Crtl_R)
        self.toolBar_2.addSeparator()
        self.toolBar_2.addAction(self.actionUndo_Ctrl_Z)
        self.toolBar_2.addAction(self.actionRedo_Ctrl_Y)
        self.toolBar_2.addAction(self.actionCut_Ctrl_X)
        self.toolBar_2.addAction(self.actionCopy_Ctrl_C)
        self.toolBar_2.addAction(self.actionPaste_Ctrl_V)

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))
        self.label.setText(_translate("MainWindow", "Device List"))
        self.pushButton.setText(_translate("MainWindow", "Finish"))
        self.menuFile_F.setTitle(_translate("MainWindow", "File (&F)"))
        self.menuEdit_E.setTitle(_translate("MainWindow", "Edit (&E)"))
        self.menuView_V.setTitle(_translate("MainWindow", "View (&V)"))
        self.menuHelp_H.setTitle(_translate("MainWindow", "Help (&H)"))
        self.toolBar.setWindowTitle(_translate("MainWindow", "toolBar"))
        self.toolBar_2.setWindowTitle(_translate("MainWindow", "toolBar_2"))
        self.actionNew_Project_Ctrl_N.setText(_translate("MainWindow", "New Project (&Ctrl+N)"))
        self.actionOpen_Project_Ctrl_O.setText(_translate("MainWindow", "Open Project (&Ctrl+O)"))
        self.actionSave_Project_Ctrl_S.setText(_translate("MainWindow", "Save Project (Ctrl+S)"))
        self.actionSettings.setText(_translate("MainWindow", "Settings &"))
        self.actionExit_X.setText(_translate("MainWindow", "Exit (&Alt+F4)"))
        self.actionUndo_Ctrl_Z.setText(_translate("MainWindow", "Undo (&Ctrl+Z)"))
        self.actionRedo_Ctrl_Y.setText(_translate("MainWindow", "Redo (&Ctrl+Y)"))
        self.actionCut_Ctrl_X.setText(_translate("MainWindow", "Cut (&Ctrl+X)"))
        self.actionCopy_Ctrl_C.setText(_translate("MainWindow", "Copy (&Ctrl+C)"))
        self.actionPaste_Ctrl_V.setText(_translate("MainWindow", "Paste (&Ctrl+V)"))
        self.actionDelete.setText(_translate("MainWindow", "Delete &"))
        self.actionSelect_All_Ctrl_A.setText(_translate("MainWindow", "Select All (Ctrl+A)"))
        self.actionDevices.setText(_translate("MainWindow", "Devices &"))
        self.actionHow_To_Use.setText(_translate("MainWindow", "How To Use"))
        self.actionReport_Issue.setText(_translate("MainWindow", "Report Issue &"))
        self.actionCheck_For_Updaes.setText(_translate("MainWindow", "Check For Updaes &"))
        self.actionAbout.setText(_translate("MainWindow", "About &"))
        self.actionReset_Crtl_R.setText(_translate("MainWindow", "Reset (&Crtl+R)"))
        self.actionHome_Ctrl_H.setText(_translate("MainWindow", "Home (&Ctrl+H)"))
        self.actionShortcut_File_Menu.setText(_translate("MainWindow", "Shortcut File Menu &"))
        self.actionShortcut_Edit_Menu.setText(_translate("MainWindow", "Shortcut Edit Menu &"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    MainWindow = QtWidgets.QMainWindow()
    ui = Ui_MainWindow()
    ui.setupUi(MainWindow)
    MainWindow.show()
    sys.exit(app.exec_())
